### DISCUSSION TITLE
Cloudlet Finishing time is longer than expected

### DISCUSSION THREAD

**Discussion by LeeBoonKong** [manoelcampos/cloudsim-plus#374] [Q&A] [answered]
I would like to ask under what circumstances or settings will cause a cloudlet to take more time to complete for the Cloudlet length and VM Speed?

For example, the speed of VM is 1000 and the length of Cloudlet is 5000, by calculation it should take 5000/1000 = 5 seconds to complete. However, during my experiment, the Cloudlet took more than 5 second to complete.

**Comment by manoelcampos** [comment, marked as answer]
Hello @LeeBoonKong 
Sorry for the late reply. Your question just got lost here.
The cloudlet execution time depends on multiple factors.
The scenario you presented is correct only if: (i) you use a `UtilizationModelFull` for Cloudlet CPU and (ii) the number of Cloudlets is less or equal to the number of virtual CPUs (vCPUs) available (regardless of the CloudletScheduler used).

If you use a `UtilizationModelDynamic` that indicates Cloudlets use just 50% of CPU all the time, they will take double of the time you showed to finish. If you have more Cloudlets than vCPUS and you are using a CloudletSchedulerTimeShared, Cloudlets have to wait to use CPU, adding up to the finish time.
